py/src/ai/h2o/sparkling/VersionComponents.py

```diff
@@ -33,12 +33,12 @@ def parseFromSparklingWaterVersion(version):
 
     @staticmethod
     def parseFromPySparkVersion(version):
-        match = re.search(r"^((\d+)\.(\d+))\.(\d+)(\.([0-9A-Za-z]+))?$", version)
+        match = re.search(r"^((\d+)\.(\d+))\.(\d+)([\\.\\+]+([0-9A-Za-z\\+\\.]+))?$", version)
```

Comment:
WDYT about simplifying the part `([\\.\\+]+([0-9A-Za-z\\+\\.]+))?` of the regular expression to `(.+)?`? Is the suffix used somewhere in the code? Can the suffix start with delimiter `.`,`+`?

I'm just wondering how to generalize the solution for all possible suffixes of spark custom builds ( '-SNAPSHOT', '.cloudera3', etc.)

Reply:
That's good point, we don't use the suffix at all at this moment

Reply:
Maybe we can just collect anything after the patch version as suffix, what do you think?

Reply:
Sounds good to me!